content/doc/developer/publishing/preparation.adoc

```diff
@@ -0,0 +1,31 @@
+---
+title: Before you start
+summary: Read this before you start
+layout: developerguide
+---
+
+== Look for similar plugins
+
+We welcome contributions from anyone, but for the benefit of Jenkins users we ask that you look for plugins solving the same or similar problems to see whether you can join forces with existing maintainers.
+You can find an overview of existing plugins on https://plugins.jenkins.io. Feel free to ask on the link:/mailing-lists[jenkinsci-dev mailing list] to see whether anyone is aware of another plugin implementing something similar to what you're planning to do.
+
+
+== Naming Convention
+
+Make sure your plugin follows link:../style-guide/#plugin-naming-convention[the naming conventions].
+
+
+== License
+
+All plugins distributed by the Jenkins project need to be free and open source software.
+This applies to the plugin source code as well as all its dependencies.
+Make sure to specify the license both in the +pom.xml+ file, as well as a +LICENSE+ file in your repository.
+
+We recommend use of the MIT license which is used for Jenkins core and most plugins and libraries, but any OSI-approved open source license will do.
+
+
+== Signup Required
+
+Jenkins plugins distributed via Jenkins project update sites need to be hosted in the https://github.com/jenkinsci[jenkinsci GitHub organization], so you will need a user account on GitHub.
+
+To actually release your plugin, you will need a https://accounts.jenkins.io[Jenkins community account] that will give you access to the https://issues.jenkins-ci.org/[issue tracker], https://wiki.jenkins-ci.org/[wiki], and https://repo.jenkins-ci.org/[Maven repository].
```

Comment:
should we remind here about the tech requirement/limitation to log at least once into Artifactory?

Reply:
Seems redundant, as that's covered in the upload permissions checklist / instructions.